app/views/users/index.html.erb

```diff
@@ -13,11 +13,9 @@
 
 <% @users.each do |user| %>
   <tr>
-    <td><%= user.name %></td>
+    <td><%= link_to user.name, user_path(user) %></td>
     <td><%= user.email %></td>
-    <td><%= link_to 'Show', user_path(user) %></td>
-    <td><%= link_to 'Edit', edit_user_path(user) %></td>
-    <td><%= link_to 'Impersonate', impersonate_user_path(user), method: 'post' %></td>
+    <td><%= link_to 'Impersonate', impersonate_user_path(user), method: 'post', class: "btn btn-danger" %></td>
```

Comment:
No, it's not dangerous; actually, it's less dangerous than the normal views an admin has.  (a) An admin can already mutate anything. (b) Impersonating a user takes the admin to the user's homepage, where nothing happens immediately.  (c) While impersonating, the only thing the admin can do is submit a homework on behalf of the user, or withdraw from the course -- and that's already a scary red button. (d) While impersonating, the admin can't see the student's unpublished scores, or edit their grades.